spring-cloud-starter-stream-source-sftp/src/main/java/org/springframework/cloud/stream/app/sftp/source/SftpSourceConfiguration.java

```diff
@@ -91,7 +120,27 @@ else if (properties.getFilenameRegex() != null) {
 		filterChain.addFilter(new SftpPersistentAcceptOnceFileListFilter(new SimpleMetadataStore(), "sftpSource"));
 
 		IntegrationFlowBuilder flowBuilder;
-		if (!properties.isStream()) {
+
+		if (properties.isStream()) {
+			SftpStreamingInboundChannelAdapterSpec messageSourceStreamingSpec =
+					Sftp.inboundStreamingAdapter(this.sftpTemplate)
+							.remoteDirectory(properties.getRemoteDir())
+							.remoteFileSeparator(properties.getRemoteFileSeparator())
+							.filter(filterChain);
+
+			flowBuilder = FileUtils.enhanceStreamFlowForReadingMode(
+					IntegrationFlows.from(messageSourceStreamingSpec,
+							properties.isDeleteRemoteFiles() ? consumerSpecWithDelete() : consumerSpec()),
+					fileConsumerProperties);
+		}
+		else if (properties.isListOnly() || properties.isTaskLauncherOutput()) {
+			return IntegrationFlows.from(sftpInboundMessageSource(), consumerSpec())
+					.handle(sftpGatewayMessageHandler(sftpSessionFactory))
```

Comment:
I think we can consider to use `Sftp.outboundGateway()` instead of direct `SftpOutboundGateway` usage.